pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -123,15 +123,15 @@ public void initialize(URL location, ResourceBundle resources) {
 
         xpathExpressionArea.plainTextChanges()
                            .map(m -> {
-                               Integer a = xpathExpressionArea.getText().lastIndexOf("/", m.getInsertionEnd());
-                               String s = xpathExpressionArea.getText().substring(a + 1);
-                               return Tuples.t(a, s);
+                               int indexOfSlash = reverseStringSearch('/', xpathExpressionArea.getText(), m.getInsertionEnd());
+                               String input = xpathExpressionArea.getText().substring(indexOfSlash + 1, m.getInsertionEnd());
+                               return Tuples.t(indexOfSlash, input, m.getInsertionEnd());
                            })
                            .filter(t -> t._2.matches("[a-zA-Z]+"))
-                           .subscribe(e -> autoComplete(e._1, e._2));
+                           .subscribe(e -> autoComplete(e._1, e._2, e._3));
     }
 
-    private void autoComplete(int slashPosition, String input) {
+    private void autoComplete(int slashPosition, String input, int endOfInsertion) {
```

Comment:
By construction, `endOfInsertion == slashPosition + input.length()` so you can drop the parameter